tracer/src/Datadog.Trace/Activity/DuckTypes/IActivity.cs

```diff
@@ -15,6 +15,8 @@ internal interface IActivity : IDuckType
     {
         string Id { get; }
 
+        string ParentId { get; }
```

Comment:
It looks like the property `Id` will be null before the Activity is started but will be guaranteed to be non-null once it is started. Since we only access the Activities after they have started, I think it's safe to assume `Id` will always be null. ✅ 

For property `RootId`, the code comments state that it may be null if both Id and ParentId are null. Since we are assuming that Id will be non-null in our cases, I think it's safe to also say that RootId will be non-null ✅ 

For property `ParentId`, you are definitely right that this may be null so I'll add the nullable reference type here 📝

Reply:
Actually, maybe we should mark these all as nullable, but in our Activity handlers we can treat them as never having a `null` value. Would that handling be preferable?

Reply:
Sounds good. I like the first one better as well, I'll just add comments for `Id` and `RootId` with comments describing _why_ we're treating them as non-null for future readers

Reply:
Updated in 49b3a80cc7a194c6d1e06819393d720842bfb864 . Feel free to resolve if it looks good to you